File: 2024/c.py
#!/usr/bin/env python3

# ICPC Domestic Qualifier
# Problem C - Honeycomb Distance

def solve(x, y):
    # normalize
    if x < 0:
        x, y = -x, -y
    if y >= 0:
        return x + y
    if x + y >= 0:
        return x
    return abs(y)

n = int(input())
for t in range(n):
    x, y = map(int, input().split())
    print(solve(x, y))

File: 2024/c.hs
#!/usr/bin/env runghc

-- ICPC Domestic Qualifier
-- Problem C - Honeycomb Distance

import Control.Monad (replicateM_)
import Control.Applicative ((<$>))

main :: IO ()
main = do
    n <- read <$> getLine :: IO Int
    replicateM_ n $ do
        [x,y] <- map read . words <$> getLine :: IO [Int]
        print $ honeycomb x y

honeycomb :: Int -> Int -> Int
honeycomb x y | x <  0     = honeycomb (-x) (-y)
              | y >= 0     = x + y
              | x + y >= 0 = x
              | otherwise  = -y
